src/services/contracts/exchange-proxy.service.ts

```diff
@@ -0,0 +1,88 @@
+import exchangeProxyAbi from '@/lib/abi/ExchangeProxy.json';
+import configs from '@/lib/config';
+import { SwapToken } from '../swap/swap.service';
+import { Swap } from '@balancer-labs/sor/dist/types';
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { contractCaller } from './contract-caller.service';
+
+export default class ExchangeProxyService {
+  network: string;
+  address: string;
+  abi: any;
+  signer: any;
+  contract: any;
+  contractWithSigner: any;
+
+  constructor() {
+    this.network = 'homestead';
+    this.address = configs[this.network].addresses.exchangeProxy;
```

Comment:
lets grab the network/address right before the usage (because the app will switch networks soon)